### PR TITLE
Add forward-only hypot support via NNGraph and torch_nntile

### PR DESCRIPTION
**Pull request by Muxas**
## Summary

Adds forward-only element-wise `hypot` support at the NNGraph and torch_nntile layers, building on the existing TensorGraph/kernel implementation.

Computes `hypot(alpha * x, beta * y)` per element (defaults `alpha=beta=1.0`, matching `torch.hypot`). Backward is intentionally **not** implemented and throws at runtime.

### NNGraph layer
- New `NNHypotOp` and `nntile::hypot(x, y, alpha, beta)` wrapper
- `backward()` throws `std::runtime_error("hypot backward is not implemented")`

### Python API
- Exposes `nntile.nn.hypot(x, y, alpha=1.0, beta=1.0)`

### torch_nntile
- Registers `aten::hypot` and `aten::hypot.out` for the nntile device
- Routes through `tensor::hypot` via the TensorGraph recorder

## Tests

- C++ NNGraph tests: structure, forward execution, backward throw, PyTorch parity (`tests_nn_ops_hypot` — all 4 cases pass)
- Python parity tests for `torch.hypot` on nntile tensors (`torch_nntile/tests/test_hypot_parity.py`)

## Notes

- No new kernel/core code — reuses existing `tensor::hypot` stack
- Autograd/training use cases requiring hypot gradients are out of scope for this PR

<div><a href="https://cursor.com/agents/bc-d6b6b328-b7b2-43df-a505-93c7c9ada247"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-d6b6b328-b7b2-43df-a505-93c7c9ada247"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

**Comment by Muxas** [comment]
@cursor review
